cupy/core/raw.pyx

```diff
@@ -105,6 +105,36 @@ cdef class RawKernel:
             self._kernel_cache[dev] = ker
         return ker
 
+    # It is not possible to implement __reduce__ for a cdef class. The
+    # two-tuple return cannot handle the keyword-only arguments, and
+    # the three-tuple return (for updating the object's internal state)
+    # does not work either, because cdef classes by default does not have
+    # __dict__. Therefore, the only way to handle keyword-only arguments
+    # for picking a cdef class is to define the following two special
+    # functions, which is in fact preferred over __reduce__.
+
+    def __getstate__(self):
+        cdef dict args = {'code': self.code,
+                          'name': self.name,
+                          'options': self.options,
+                          'backend': self.backend,
+                          'translate_cucomplex': self.translate_cucomplex,
+                          'file_path': self.file_path,
+                          'name_expressions': self.name_expressions}
+        args['enable_cooperative_groups'] = self.enable_cooperative_groups
```

Comment:
Nitpicks: why not set it in dict constructor?
```suggestion
                          'name_expressions': self.name_expressions,
                          'enable_cooperative_groups': self.enable_cooperative_groups}
```

Reply:
Yeah I'd love to do that, but it'd go over the 79 characters length limit 😞 Can I append `noqa` to fix it?

Reply:
How about starting a newline after `cdef dict args = {`?

Reply:
How about writing as follows?
```py
return {
    'code': self.code,
    ...
}
```

Reply:
Ah, it might work, let me give it a shot!

Reply:
Done in 07da1b2. @asi1024 PTAL.